cmd/git/git.go

```diff
@@ -553,17 +561,46 @@ type DSGit struct {
 	// PairProgramming mode
 	PairProgramming bool
 	// CommitsHash is a map of commit hashes for each repo
-	CommitsHash map[string]map[string]struct{}
+	CommitsHash     map[string]map[string]struct{}
 	// Publisher & stream
 	Publisher
-	Stream string // stream to publish the data
+	Stream          string // stream to publish the data
+	Logger          logger.Logger
 }
 
 // AddPublisher - sets Kinesis publisher
 func (j *DSGit) AddPublisher(publisher Publisher) {
 	j.Publisher = publisher
 }
 
+func (j *DSGit) AddLogger(ctx *shared.Ctx) {
+	client, err := elastic.NewClientProvider(&elastic.Params{
+		URL:      ctx.ESURL,
+	})
+	if err != nil {
+		shared.Printf("AddLogger error: %+v", err)
+		return
+	}
+	logProvider, err := logger.NewLogger(client, os.Getenv("ENV"))
+	if err != nil {
+		shared.Printf("AddLogger error: %+v", err)
+		return
+	}
+	j.Logger = *logProvider
+}
+
+func (j *DSGit) WriteLog(ctx *shared.Ctx, status, message string) {
+	_ = j.Logger.Write(&logger.Log{
+		Connector:     GitDataSource,
+		Configuration: []map[string]string{{"REPO_URL": j.URL, "ES_URL": ctx.ESURL}},
+		Status:        status,
+		CreatedAt:     time.Now(),
+		UpdatedAt:     time.Now(),
+		ProjectSlug:   ctx.Project,
```

Comment:
project slug should be in configuration slice, you will need to update shared version

Reply:
done